### PR TITLE
Remove NumPy from Callback scripts

### PR DESCRIPTION
**Pull request by ishandutta0098**
## What does this PR do?
This PR is a part of #16649 

I have replaced the use of `np.Inf` and `np.nan` with `float('inf')` and `float('nan')`  for the following scripts:
- `model_checkpoint.py`
- `early_stopping.py`
- `test_early_stopping.py`

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

Requesting Review: @carmocca @Borda

### CODE REVIEW

**Review by aniketmaurya — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/lightning/pytorch/callbacks/early_stopping.py [hunk 1]
```diff
@@ -123,7 +122,7 @@ def __init__(
             raise MisconfigurationException(f"`mode` can be {', '.join(self.mode_dict.keys())}, got {self.mode}")
 
         self.min_delta *= 1 if self.monitor_op == torch.gt else -1
-        torch_inf = torch.tensor(np.Inf)
+        torch_inf = torch.tensor(float("inf"))
```
Comment (awaelchli) [inline src/lightning/pytorch/callbacks/early_stopping.py:125]: You could also do:
```suggestion
        torch_inf = torch.tensor(torch.inf)
```
which perhaps is a tiny bit cleaner to read (same for nan).

**Review by awaelchli — APPROVED** [review]

File: src/lightning/pytorch/callbacks/early_stopping.py (hunk 1 above)
Comment (ishandutta0098) [reply src/lightning/pytorch/callbacks/early_stopping.py]: I have replaced the use of float with torch

**Review by dingusagar — APPROVED** [review]
